### PR TITLE
create retrieval_word_overlap.py

### PR DESCRIPTION
**Pull request by haniesedghi**
This code implements a word_overlap retrieval baseline. given a question and a file consisting all the knowledge, e.g. list of all table row sentences. 
In order to score the retrieval model, picks a random list of question, answer pairs (e.g. from AristoQuiz dataset) and it checks if the answer exists in the topk retrieved sentences.
btw, the code has been checked by Joel

### CODE REVIEW

File: deep_qa/contrib/background_search/retrieval_word_overlap [hunk 1]
```diff
@@ -0,0 +1,285 @@
+"""
```
Comment (matt-gardner) [inline deep_qa/contrib/background_search/retrieval_word_overlap]: Can you rename this file to `retrieval_word_overlap.py`?  It'll make github show it correctly as a python file, and will make this a whole lot easier to read.

**Comment by matt-gardner** [comment]
@DeNeutoy, are you looking at this?  I was holding off, because Hanie requested your review.

**Comment by DeNeutoy** [comment]
Sorry I missed this, LGTM.

**Comment by matt-gardner** [comment]
Once the pylint errors are fixed and the tests pass, feel free to merge.

**Pull request merged**